demo/app-root.ts

```diff
@@ -1,20 +1,286 @@
-import { html, css, LitElement } from 'lit';
-import { customElement } from 'lit/decorators.js';
-import '../src/your-webcomponent';
+import { SearchService } from '@internetarchive/search-service';
+import { LocalCache } from '@internetarchive/local-cache';
+import { html, css, LitElement, PropertyValues } from 'lit';
+import { customElement, query, state } from 'lit/decorators.js';
+import { SharedResizeObserver } from '@internetarchive/shared-resize-observer';
+import { CollectionNameCache } from '@internetarchive/collection-name-cache';
+import type { CollectionBrowser } from '../src/collection-browser';
+import '../src/collection-browser';
 
 @customElement('app-root')
 export class AppRoot extends LitElement {
+  private searchService = SearchService.default;
+
+  private resizeObserver = new SharedResizeObserver();
+
+  private localCache = new LocalCache();
+
+  private collectionNameCache = new CollectionNameCache({
+    searchService: this.searchService,
+    localCache: this.localCache,
+  });
+
+  @state() private currentPage?: number;
+
+  @state() private searchQuery?: string;
+
+  @state() private cellWidth: number = 18;
+
+  @state() private cellHeight: number = 29;
+
+  @state() private rowGap: number = 1.7;
+
+  @state() private colGap: number = 1.7;
+
+  @query('#base-query-field') private baseQueryField!: HTMLInputElement;
+
+  @query('#page-number-input') private pageNumberInput!: HTMLInputElement;
+
+  @query('collection-browser') private collectionBrowser!: CollectionBrowser;
+
+  private searchPressed(e: Event) {
+    e.preventDefault();
+    this.searchQuery = this.baseQueryField.value;
+    if ((this.currentPage ?? 1) > 1) {
+      this.collectionBrowser.goToPage(this.currentPage ?? 1);
+    }
+    this.currentPage = 1;
```

Comment:
QUESTION: Is it correct that `this.currentPage` should always be set here to 1, even if it entered the function with currentPage set to something else, e.g., 5?

Reply:
Ah yeah this is some old code in the demo app. Removed.